main.tf

```diff
@@ -27,16 +27,16 @@ resource "aws_sns_topic" "this" {
 }
 
 resource "aws_sns_topic_subscription" "this" {
-  for_each = local.enabled ? var.subscribers : {}
+  for_each = { for subscription in toset(var.subscribers) : subscription.endpoint => subscription if local.enabled }
 
   topic_arn              = join("", aws_sns_topic.this.*.arn)
-  protocol               = var.subscribers[each.key].protocol
-  endpoint               = var.subscribers[each.key].endpoint
-  endpoint_auto_confirms = var.subscribers[each.key].endpoint_auto_confirms
-  raw_message_delivery   = var.subscribers[each.key].raw_message_delivery
+  protocol               = each.value.protocol
+  endpoint               = each.value.endpoint
+  endpoint_auto_confirms = lookup(each.value, "endpoint_auto_confirms", false)
+  raw_message_delivery   = lookup(each.value, "raw_message_delivery", false)
   redrive_policy = var.sqs_dlq_enabled ? coalesce(var.redrive_policy, jsonencode({
     deadLetterTargetArn = join("", aws_sqs_queue.dead_letter_queue.*.arn)
-    maxReceiveCount     = var.redrive_policy_max_receiver_count
+    # maxReceiveCount     = var.redrive_policy_max_receiver_count
```

Comment:
keeping this key results in the following error: 

```
│ Error: error setting SNS Topic Subscription (arn:aws:sns:us-west-2:<account-id>:eg-test-sns.fifo:310f9002-0630-42dc-ba81-1ce9521840d3) attribute (RedrivePolicy): InvalidParameter: Invalid parameter: RedrivePolicy: invalid JSON. Attribute maxReceiveCount is unknown.
```

The only reference I could find for it in the AWS api docs was [here](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SetQueueAttributes.html), its suspiciously not called out in the [boto3 documentation for SNS topic subscriptions](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sns.html#SNS.Client.subscribe) nor in the [api docs for SNS Subscribe](https://docs.aws.amazon.com/sns/latest/api/API_Subscribe.html)

Reply:
Can we set it to `null` if it's not supplied ? or default the input to `null`?

Reply:
sure can, updated